qiskit/ignis/verification/entanglement/parallelize.py

```diff
@@ -0,0 +1,449 @@
+        '''
+
+        a = self.backend.properties().to_dict()['gates']
+        cxerrordict = {}
+        for i in a:
+            if len(i['qubits']) == 1:
+                continue
+            if len(i['qubits']) == 2:
+                b = tuple(i['qubits'])
+                if b in cxerrordict.keys():
+                    pass
+                cxerrordict[b] = i['parameters'][0]['value']
+                if (b[1], b[0]) not in cxerrordict.keys():
+                    cxerrordict[(b[1], b[0])] = i['parameters'][0]['value']
+                else:
+                    pass
+
+        newcxerrordict = cxerrordict
+
+#         for a in list(cxerrordict.keys()):
+#             if (a[1],a[0]) not in cxerrordict.keys():
+#                 newcxerrordict[(a[1],a[0])] = cxerrordict[a]
+        return newcxerrordict
+
+    def get_cx_length(self):
+        '''
+        Gets dict of relevant CNOT gate lengths
+        '''
+
+        a = self.backend.properties().to_dict()['gates']
+        cxlengthdict = {}
+        for i in a:
+            if len(i['qubits']) == 1:
+                continue
+            if len(i['qubits']) == 2:
+                b = tuple(i['qubits'])
+                if b in cxlengthdict.keys():
+                    pass
+                cxlengthdict[b] = i['parameters'][1]['value']
+                if (b[1], b[0]) not in cxlengthdict.keys():
+                    cxlengthdict[(b[1], b[0])] = i['parameters'][1]['value']
+                else:
+                    pass
+
+        newcxlengthdict = cxlengthdict
+#         for a in list(cxlengthdict.keys()):
+#             if (a[1],a[0]) not in cxlengthdict.keys():
+#                 newcxlengthdict[(a[1],a[0])] = cxlengthdict[a]
+
+        return newcxlengthdict
+
+    def child_sorter(self, children, parent):
+        '''
+        Sorts children nodes based on error/length
+        '''
+
+        newchildren = {}
+        for i in children:
+            newchildren[(i, parent)] = self.get_cx_error()[(i, parent)]
+        ss = sorted(newchildren.items(), key=lambda x: x[1])
+        newchildrenlist = [a[0][0] for a in ss]
+        return newchildrenlist
+
+    def get_tier_dict(self):
+        '''
+        Take the nodes of the BConfig to create a Tier Dictionary,
+        where keys are the steps in the process,
+        and the values are the connections following pattern of:
+        [controlled qubit, NOT qubit]. Thus the
+        backend's GHZ state is parallelized.
+        Args:
+        Returns:
+           Tier dictionary - [step in process, control-target connection].
+            Facilitates parallelized GHZ circuits
+        '''
+
+        tier = {}
+        tierDM = {}
+        length = len(self.nodes.keys())
+        trashlist = []
+        tier[0] = (self.get_best_node())
+        tierDM[self.get_best_node()] = 0
+
+        trashlist.append(self.get_best_node())
+        parent = self.get_best_node()
+
+        for x in range(length):
+            tier[x] = [[]]
```

Comment:
You can use dictionary comprehension:

`tier = {x: [[]] for x in range(length)}`

Reply:
done